lib/vrl/stdlib/src/parse_apache_log.rs

```diff
@@ -0,0 +1,298 @@
+use crate::log_util;
+use std::collections::BTreeMap;
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ParseApacheLog;
+
+impl Function for ParseApacheLog {
+    fn identifier(&self) -> &'static str {
+        "parse_apache_log"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::BYTES,
+                required: true,
+            },
+            Parameter {
+                keyword: "format",
+                kind: kind::BYTES,
+                required: true,
+            },
+            Parameter {
+                keyword: "timestamp_format",
+                kind: kind::BYTES,
+                required: false,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let variants = vec![value!("common"), value!("combined"), value!("error")];
+
+        let value = arguments.required("value");
+        let format = arguments
+            .required_enum("format", &variants)?
+            .try_bytes()
+            .expect("format not bytes");
+
+        let timestamp_format = arguments.optional("timestamp_format");
+
+        Ok(Box::new(ParseApacheLogFn {
+            value,
+            format,
+            timestamp_format,
+        }))
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "parse apache common log",
+                source: r#"encode_json(parse_apache_log!(s'127.0.0.1 bob frank [10/Oct/2000:13:55:36 -0700] "GET /apache_pb.gif HTTP/1.0" 200 2326', "common"))"#,
+                result: Ok(
+                    indoc! {r#"s'{"host":"127.0.0.1","identity":"bob","message":"GET /apache_pb.gif HTTP/1.0","method":"GET","path":"/apache_pb.gif","protocol":"HTTP/1.0","size":2326,"status":200,"timestamp":"2000-10-10T20:55:36+00:00","user":"frank"}'"#},
```

Comment:
Since we unfortunately seem to hit the same limitation we have in other places to use timestamp literals in the asserted result, `indoc` could be omitted here.

Reply:
This is fixed in https://github.com/timberio/vector/pull/6594